dashboard/app/mailers/pd/workshop_mailer.rb

```diff
@@ -228,6 +228,13 @@ def save_timestamp
     Pd::EnrollmentNotification.create(enrollment: @enrollment, name: action_name)
   end
 
+  # Virtual workshops should not have any mail sent.
+  def check_should_send
+    if @workshop&.subject&.include? "Virtual"
+      mail.perform_deliveries = false
```

Comment:
Is this the "right" way to do this?  [Action Mailer docs](https://guides.rubyonrails.org/action_mailer_basics.html#action-mailer-configuration) describe `perform_deliveries` this way:

> Determines whether deliveries are actually carried out when the deliver method is invoked on the Mail message. By default they are, but this can be turned off to help functional testing.

That sounds a little like this feature is designed for testing, not for production logic.  But then there's an example [in this other part of the docs](https://guides.rubyonrails.org/action_mailer_basics.html#action-mailer-callbacks):

```rb
    def prevent_delivery_to_guests
      if @user && @user.guest?
        mail.perform_deliveries = false
      end
    end
```

So I guess it's fine? 🤷‍♂ 

One reason I'm questioning this is the need for many safe-navigation operators.  Should this logic live in a context where we're sure there's a workshop involved?

Reply:
Yeah, I had similar doubts until I found that sample code which seemed to imply it was safe for regular use.

In production, 173 of 6094 workshops have a `nil` subject, but from a code inspection it appears we should always have a workshop, so I'll relax the safe-navigation a little here.